Updated code for moveLeft function to address issue where code not executed

TheNephalim opened the pull request:
updated the code for moveLeft - moved the code calling the beforeMoveLeft above the call to the main moveLeft callback. Removed the return before the call to self.callbacks.moveToLeft otherwise the rest of the code in the method is not executed.

Commit:
the name of the initialization method conflicted with another library that used the same name; changed the initialization method name so it no longer conflicted with the other library.
updated the code for moveLeft - moved the code calling the beforeMoveLeft above the call to the main moveLeft callback.  Removed the return before the call to self.callbacks.moveToLeft otherwise the rest of the code in the method is not executed.

Commit:
updated the code for moveLeft - moved the code calling the beforeMoveLeft above the call to the main moveLeft callback. Removed the return before the call to self.callbacks.moveToLeft otherwise the rest of the code in the method is not executed.

crlcu commented:
Please run the gulp command as well so the minified version gets updated.
Thanks

Commit:
ran gulp command to generate minified version of the library

Commit:
file updated after running npm install

TheNephalim commented:
Ran the gulp command and generated the minified version of the library.

TheNephalim commented:
Is there anything else I need to do?

crlcu commented:
@TheNephalim could you fix the conflicts please for the package-lock.json ?

Commit:
updated the variables for setting whether the browser isSafari or IE - the variable isSafari was picking up a chrome browser as Safari
if the intent is determine whether the browser is Chrome, there should be a separate variable like isChrome indicating the browser is a Chrome browser
updated the fireSearch to always return true - considering getting rid of this method and the logic that uses it
prefer to have it fire on any number of characters entered and not just two
updated the code that searches the options to use native JavaScript - the logic in jQuery that searches for the options added an unacceptable level of overhead
when the number of options went into the thousands; tested using a data set having close to 8000 options and the search function was very slow
using native JavaScript instead of jQuery sped up the search funtion dramatically
these updates are found in the methods filterItems and unfilterItems
added a method searchItem that uses indexOf instead of RegExp because it is faster than RegExp
added showItem and hideItem that duplicates the functionality of mHide and mShow using vanilla JavaScript instead of jQuery
also added unfilterOptionGroup and filterOptionGroups that also use vanilla JavaScript instead of jQuery and a slightly different logic - i.e., the methods are
self contained
Updated the events function that handles the calls for filtering/unfiltering options to use the newly introduced methods

Commit:
removed gulp-uglify from the package.json
installed gulp-terser

Commit:
updated the file to remove references to gulp-uglify
updated code to use gulp-terser

Commit:
ran gulp to minify the multiselect.js

Commit:
updated the check for safari browsers

Commit:
upgraded standard-version to version 8.0.1

TheNephalim committed:
Merge branch 'master' into master
